querydsl-mongodb/src/main/java/com/querydsl/mongodb/MongodbSerializer.java

```diff
@@ -138,11 +138,9 @@ public Object visit(Operation<?> expr, Void context) {
             return asDBObject("$or", list);
 
         } else if (op == Ops.NE) {
-            if (isReference(expr, 0)) {
-                return asDBObject(asDBKey(expr, 0), asDBObject("$ne", asReference(expr, 1)));
-            } else {
-                return asDBObject(asDBKey(expr, 0), asDBObject("$ne", asDBValue(expr, 1)));
-            }
+            Path<?> arg1 = (Path<?>) expr.getArg(0);
+            Constant<?> arg2 = (Constant<?>) expr.getArg(1);
```

Comment:
I would rename these variables to `path` and `constant` to be consistent with your changes on lines 198 and 215.